src/main/java/org/zalando/nakadi/service/ConsumerLimitingCleaningService.java

```diff
@@ -0,0 +1,141 @@
+package org.zalando.nakadi.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.KeeperException;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.Hours;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+import org.zalando.nakadi.repository.zookeeper.ZooKeeperHolder;
+
+import java.nio.charset.Charset;
+
+import static org.zalando.nakadi.service.ConsumerLimitingService.CONNECTIONS_ZK_PATH;
+
+@Service
+public class ConsumerLimitingCleaningService {
+
+    public static final String CLEANING_ZK_PATH = "/nakadi/consumers/cleaning";
+
+    private static final Charset CHARSET_UTF8 = Charset.forName("UTF-8");
+    private static final Logger LOG = LoggerFactory.getLogger(ConsumerLimitingCleaningService.class);
+    private static final int HANGING_NODES_CLEAN_PERIOD_H = 6;
+
+    private final ZooKeeperHolder zkHolder;
+    private final ObjectMapper objectMapper;
+    private final ConsumerLimitingService limitingService;
+
+    @Autowired
+    public ConsumerLimitingCleaningService(final ZooKeeperHolder zkHolder, final ObjectMapper objectMapper,
+                                           final ConsumerLimitingService limitingService) {
+        this.zkHolder = zkHolder;
+        this.objectMapper = objectMapper;
+        this.limitingService = limitingService;
+    }
+
+    @Scheduled(fixedRate = HANGING_NODES_CLEAN_PERIOD_H * 3600000)
```

Comment:
Maybe just put the multiplication in the constant and use it here with name like PERIOD_HANGING_NODES_CLEAN.

PERIOD_HANGING_NODES_CLEAN = 6 * 60 * 60 * 1000;